### ISSUE TITLE
Linking error while building on Windows

### ISSUE THREAD

**Issue by cosmicspacedragon** [Cldfire/nvml-wrapper#10]
I keep getting this error when I try to compile a crate with this crate as a dependency:
```
error: linking with `link.exe` failed: exit code: 1181
  |
  = note: "C:\\Program Files (x86)\\Microsoft Visual Studio\\2017\\Community\\VC\\Tools\\MSVC\\14.16.27023\\bin\\HostX64\\x64\\link.exe" "/NOLOGO" "/NXCOMPAT" "/LIBPATH:C:\\Users\\Markus\\.rustup\\toolchains\\stable-x86_64-pc-windows-msvc\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib" "D:\\Programming\\Rust\\active\\nvdup\\target\\release\\deps\\nvdup-766bc952328f798a.nvdup.6dghzu5s-cgu.0.rcgu.o" "D:\\Programming\\Rust\\active\\nvdup\\target\\release\\deps\\nvdup-766bc952328f798a.nvdup.6dghzu5s-cgu.1.rcgu.o" "D:\\Programming\\Rust\\active\\nvdup\\target\\release\\deps\\nvdup-766bc952328f798a.nvdup.6dghzu5s-cgu.10.rcgu.o" "D:\\Programming\\Rust\\active\\nvdup\\target\\release\\deps\\nvdup-766bc952328f798a.nvdup.6dghzu5s-cgu.11.rcgu.o" "D:\\Programming\\Rust\\active\\nvdup\\target\\release\\deps\\nvdup-766bc952328f798a.nvdup.6dghzu5s-cgu.12.rcgu.o" "D:\\Programming\\Rust\\active\\nvdup\\target\\release\\deps\\nvdup-766bc952328f798a.nvdup.6dghzu5s-cgu.13.rcgu.o" "D:\\Programming\\Rust\\active\\nvdup\\target\\release\\deps\\nvdup-766bc952328f798a.nvdup.6dghzu5s-cgu.14.rcgu.o" "D:\\Programming\\Rust\\active\\nvdup\\target\\release\\deps\\nvdup-766bc952328f798a.nvdup.6dghzu5s-cgu.15.rcgu.o" "D:\\Programming\\Rust\\active\\nvdup\\target\\release\\deps\\nvdup-766bc952328f798a.nvdup.6dghzu5s-cgu.2.rcgu.o" "D:\\Programming\\Rust\\active\\nvdup\\target\\release\\deps\\nvdup-766bc952328f798a.nvdup.6dghzu5s-cgu.3.rcgu.o" "D:\\Programming\\Rust\\active\\nvdup\\target\\release\\deps\\nvdup-766bc952328f798a.nvdup.6dghzu5s-cgu.4.rcgu.o" "D:\\Programming\\Rust\\active\\nvdup\\target\\release\\deps\\nvdup-766bc952328f798a.nvdup.6dghzu5s-cgu.5.rcgu.o" "D:\\Programming\\Rust\\active\\nvdup\\target\\release\\deps\\nvdup-766bc952328f798a.nvdup.6dghzu5s-cgu.6.rcgu.o" "D:\\Programming\\Rust\\active\\nvdup\\target\\release\\deps\\nvdup-766bc952328f798a.nvdup.6dghzu5s-cgu.7.rcgu.o" "D:\\Programming\\Rust\\active\\nvdup\\target\\release\\deps\\nvdup-766bc952328f798a.nvdup.6dghzu5s-cgu.8.rcgu.o" "D:\\Programming\\Rust\\active\\nvdup\\target\\release\\deps\\nvdup-766bc952328f798a.nvdup.6dghzu5s-cgu.9.rcgu.o" "/OUT:D:\\Programming\\Rust\\active\\nvdup\\target\\release\\deps\\nvdup-766bc952328f798a.exe" "D:\\Programming\\Rust\\active\\nvdup\\target\\release\\deps\\nvdup-766bc952328f798a.2l1vb010stq23k08.rcgu.o" "/OPT:REF,ICF" "/DEBUG" "/NATVIS:C:\\Users\\Markus\\.rustup\\toolchains\\stable-x86_64-pc-windows-msvc\\lib\\rustlib\\etc\\intrinsic.natvis" "/NATVIS:C:\\Users\\Markus\\.rustup\\toolchains\\stable-x86_64-pc-windows-msvc\\lib\\rustlib\\etc\\liballoc.natvis" "/NATVIS:C:\\Users\\Markus\\.rustup\\toolchains\\stable-x86_64-pc-windows-msvc\\lib\\rustlib\\etc\\libcore.natvis" "/LIBPATH:D:\\Programming\\Rust\\active\\nvdup\\target\\release\\deps" "/LIBPATH:C:\\Program Files\\NVIDIA Corporation\\NVSMI" "/LIBPATH:C:\\Users\\Markus\\.rustup\\toolchains\\stable-x86_64-pc-windows-msvc\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib" "D:\\Programming\\Rust\\active\\nvdup\\target\\release\\deps\\libnvml_wrapper-04a79d4e90b7683a.rlib" "D:\\Programming\\Rust\\active\\nvdup\\target\\release\\deps\\libnvml_wrapper_sys-b4e13e775667842f.rlib" "D:\\Programming\\Rust\\active\\nvdup\\target\\release\\deps\\libbitflags-ad952368103d65d7.rlib" "D:\\Programming\\Rust\\active\\nvdup\\target\\release\\deps\\liberror_chain-ba0720c77fe09a14.rlib" "D:\\Programming\\Rust\\active\\nvdup\\target\\release\\deps\\libbacktrace-9b8e9d1a1fee94a6.rlib" "D:\\Programming\\Rust\\active\\nvdup\\target\\release\\deps\\librustc_demangle-44d12e35ab5684f9.rlib" "D:\\Programming\\Rust\\active\\nvdup\\target\\release\\deps\\libcfg_if-14e084dd51270587.rlib" "D:\\Programming\\Rust\\active\\nvdup\\target\\release\\deps\\libwinapi-adf1e76229356873.rlib" "C:\\Users\\Markus\\.rustup\\toolchains\\stable-x86_64-pc-windows-msvc\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib\\libstd-c1e537280a7eb2d9.rlib" "C:\\Users\\Markus\\.rustup\\toolchains\\stable-x86_64-pc-windows-msvc\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib\\libpanic_unwind-fea7faaed3d25759.rlib" "C:\\Users\\Markus\\.rustup\\toolchains\\stable-x86_64-pc-windows-msvc\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib\\libbacktrace_sys-b0e97dc981603010.rlib" "C:\\Users\\Markus\\.rustup\\toolchains\\stable-x86_64-pc-windows-msvc\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib\\libunwind-d93e4acd3f7f9acb.rlib" "C:\\Users\\Markus\\.rustup\\toolchains\\stable-x86_64-pc-windows-msvc\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib\\librustc_demangle-42249eebbc8b7bf7.rlib" "C:\\Users\\Markus\\.rustup\\toolchains\\stable-x86_64-pc-windows-msvc\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib\\liblibc-9bb7faeb8ad341e3.rlib" "C:\\Users\\Markus\\.rustup\\toolchains\\stable-x86_64-pc-windows-msvc\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib\\liballoc-ae07d6dbc61a1548.rlib" "C:\\Users\\Markus\\.rustup\\toolchains\\stable-x86_64-pc-windows-msvc\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib\\librustc_std_workspace_core-53d2dfe88d5ede66.rlib" "C:\\Users\\Markus\\.rustup\\toolchains\\stable-x86_64-pc-windows-msvc\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib\\libcore-797cfa1fd40eb75c.rlib" "C:\\Users\\Markus\\.rustup\\toolchains\\stable-x86_64-pc-windows-msvc\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib\\libcompiler_builtins-8424507037470daf.rlib" "nvml.lib" "advapi32.lib" "dbghelp.lib" "kernel32.lib" "advapi32.lib" "ws2_32.lib" "userenv.lib" "msvcrt.lib"
  = note: LINK : fatal error LNK1181: cannot open input file 'nvml.lib'
```

I'm not sure what I'm supposed to do since the README doesn't say anything about `nvml.lib`. There seems to be no such file inside `C:\Program Files\NVIDIA Corporation`.

**Comment by Cldfire** [comment]
Ooo, yes. So, the `nvml.lib` file that the linker is complaining about not being able to find is something that I've generated and added to the repo to be provided with the crate. The issue is, the commits for that are on the `devel` branch at the moment.

In the short-term, you can resolve this by depending on the `devel` branch from this repo:

```toml
[dependencies]
nvml-wrapper = { git = "https://github.com/Cldfire/nvml-wrapper.git", branch = "devel" }
```

Obviously that's a very ugly solution long-term. I am slowly but surely working my way towards having the time and motivation to tie up the loose ends on the `devel` branch and merge it into master / cut an `0.5.0` release, but I can't provide a precise timeline unfortunately.

Thanks for bringing this to my attention though, I had completely forgotten that the current crates.io release isn't compiling with the MSVC toolchain on windows. I guess it goes to show how few people use this crate ;-)

**Comment by Cldfire** [comment]
I just released 0.5.0 with the import library included, resolving this issue.
